base/linalg/uniformscaling.jl

```diff
@@ -53,6 +53,29 @@ eltype(::Type{UniformScaling{T}}) where {T} = T
 ndims(J::UniformScaling) = 2
 getindex(J::UniformScaling, i::Integer,j::Integer) = ifelse(i==j,J.λ,zero(J.λ))
 
+function getindex(x::UniformScaling{T}, n::AbstractRange{<:Integer}, m::AbstractRange{<:Integer}) where T
+    if length(n) == length(m) && step(n) == step(m)
+        k = first(n) - first(m)
+        if length(n) - abs(k) > 0
+            v = fill(x.λ, length(n) - abs(k))
+            return spdiagm(k => v)
```

Comment:
This doesn't seem right for non-unit steps.  Consider
```jl
julia> eye(10,10)[1:2:7, 2:2:8]
4×4 Array{Float64,2}:
 0.0  0.0  0.0  0.0
 0.0  0.0  0.0  0.0
 0.0  0.0  0.0  0.0
 0.0  0.0  0.0  0.0
```

Reply:
Nice catch. Should be fixed by additionally checking if `k % step(n) == 0`.